// @hint/hint-strict-transport-security/dist/tests/_common.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const mock = require("mock-require");
const utils_create_server_1 = require("@hint/utils-create-server");
const utilsNetwork = require("@hint/utils-network");
exports.OkayMaxAge = 31536000;
exports.smallMaxAge = 1;
exports.defaultMinimum = 10886400;
exports.stsHeader = 'strict-transport-security';
exports.maxAgeOnlyHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}` };
exports.includeSubDomainsHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}; includeSubDomains` };
exports.preloadHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}; includeSubDomains; preload` };
exports.mixCaseHeader = { [exports.stsHeader]: `Max-Age=${exports.OkayMaxAge}` };
exports.quotedStringHeader = { [exports.stsHeader]: `max-age="${exports.OkayMaxAge}"; includeSubDomains; preload` };
exports.tooShortHeader = { [exports.stsHeader]: `max-age=${exports.smallMaxAge}` };
exports.noMaxAgeHeader = { [exports.stsHeader]: `maxage=${exports.OkayMaxAge}; includeSubDomains; preload` };
exports.multipleMaxAgeHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}; max-age=${exports.OkayMaxAge + 1}` };
exports.multipleincludeSubDomainsHeader = { [exports.stsHeader]: `includeSubDomains; max-age=${exports.OkayMaxAge}; includeSubDomains` };
exports.wrongDelimiterHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}, includeSubDomains; preload` };
exports.includeUnitMaxAgeHeader = { [exports.stsHeader]: `max-age=${exports.OkayMaxAge}s; includeSubDomains; preload` };
exports.notPreloadableError = `www subdomain does not support HTTPS`;
exports.preloaded = { status: 'preloaded' };
exports.unknown = { status: 'unknown' };
exports.noErrors = { errors: [] };
exports.hasErrors = { errors: [{ message: exports.notPreloadableError }] };
exports.generateTooShortError = (value) => {
    return `The '${exports.stsHeader}' header 'max-age' value should be more than '${value}'.`;
};
exports.noHeaderError = `The '${exports.stsHeader}' header was not specified.`;
exports.noMaxAgeError = `The '${exports.stsHeader}' header requires 'max-age' directive.`;
exports.duplicateDirectivesError = `The '${exports.stsHeader}' header contains duplicate directives.`;
exports.tooShortErrorDefault = exports.generateTooShortError(exports.defaultMinimum);
exports.wrongFormatError = `The '${exports.stsHeader}' header has the wrong format.`;
exports.statusServiceError = `Error getting preload status.`;
exports.preloadableServiceError = `Error getting preload eligibility.`;
exports.problemWithVerificationEndpoint = `Error getting preload status.`;
exports.faviconHeaderMaxAgeOnly = {
    '/': { content: utils_create_server_1.generateHTMLPage() },
    '/favicon.ico': { headers: { [exports.stsHeader]: `max-age=${exports.OkayMaxAge + 100}` } }
};
exports.generateHTMLPageData = (content) => {
    return {
        content,
        headers: exports.maxAgeOnlyHeader
    };
};
exports.htmlPageWithScriptData = exports.generateHTMLPageData(utils_create_server_1.generateHTMLPage(undefined, '<script src="test.js"></script>'));
exports.htmlPageWithManifestData = exports.generateHTMLPageData(utils_create_server_1.generateHTMLPage('<link rel="manifest" href="test.webmanifest">'));
exports.requestJSONAsyncMock = (responseObject) => {
    const isHTTPS = () => {
        return true;
    };
    const isRegularProtocol = () => {
        return true;
    };
    const normalizeString = (str = '') => {
        return str.toLowerCase();
    };
    const requestJSONAsync = (uri) => {
        let response;
        if (uri.includes('/api/v2/preloadable')) {
            response = responseObject.preloadable;
        }
        else {
            response = responseObject.status;
        }
        if (!response) {
            return Promise.reject(new Error('Error with the verification service.'));
        }
        return Promise.resolve(response);
    };
    utilsNetwork.isRegularProtocol = isRegularProtocol;
    utilsNetwork.isHTTPS = isHTTPS;
    utilsNetwork.requestJSONAsync = requestJSONAsync;
    mock('@hint/utils-network', utilsNetwork);
    mock('@hint/utils-string', { normalizeString });
};
